src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -341,6 +341,17 @@ public final class Guardrails implements GuardrailsMBean
                                : format("The keyspace %s has a replication factor of %s, below the failure threshold of %s.",
                                         what, value, threshold));
 
+    /**
+     * Guardrail on the maximum replication factor.
+     */
+    public static final MaxThreshold maximumReplicationFactor =
+    new MaxThreshold("maximum_replication_factor",
+                  state -> CONFIG_PROVIDER.getOrCreate(state).getMaximumReplicationFactorWarnThreshold(),
+                  state -> CONFIG_PROVIDER.getOrCreate(state).getMaximumReplicationFactorFailThreshold(),
+                  (isWarning, what, value, threshold) ->
+                  format("The keyspace %s has a replication factor of %s, above the %s threshold of %s",
+                         what, value, isWarning ? "warning" : "failure", threshold));
```

Comment:
This should probably end with a period:
```suggestion
                  format("The keyspace %s has a replication factor of %s, above the %s threshold of %s.",
                         what, value, isWarning ? "warning" : "failure", threshold));
```
The new unit tests seem to expect that dot, and I think that's why some of them fail. Did you run them locally?

Reply:
sorry, missed that when I ran the tests, added the . back in.